### PR TITLE
expose LLM completions over graphql

### PR DESCRIPTION
**Pull request by coury-clark**
Adds a graphql wrapper around the completions API. This is a little easier for some clients to consume who may not care about streaming content. I just copied over the same validations from the streaming endpoint for now, I'm not sure what the long term stability of this API will be.

Just for clarity, this isn't _code completions_ - it's just copying the name of the streaming API
`.api/completions/stream`

## Test plan

the `completions` graphql query is available

image

### CODE REVIEW

File: enterprise/cmd/frontend/internal/completions/resolvers/resolver.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	client, err := streaming.GetCompletionStreamClient(completionsConfig.Provider, completionsConfig.AccessToken, completionsConfig.Model)
+	if err != nil {
+		return "", errors.Wrap(err, "GetCompletionStreamClient")
+	}
+
+	var last string
+	if err := client.Stream(ctx, convertParams(args), func(event types.CompletionEvent) error {
+		// each completion is just a partial of the final result, since we're in a sync request anyway
+		// we will just wait for the final completion event
+		fmt.Println(event.Completion)
```
Comment (coury-clark) [inline enterprise/cmd/frontend/internal/completions/resolvers/resolver.go:48]:
```suggestion
```

File: cmd/frontend/graphqlbackend/completions.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package graphqlbackend
+
+import "golang.org/x/net/context"
```
Comment (novoselrok) [inline cmd/frontend/graphqlbackend/completions.go:3]: Can this be just `import "context"`?

File: enterprise/cmd/frontend/internal/insights/init.go [hunk 3]
```diff
@@ -39,6 +40,7 @@ func Init(
 	}
 	enterpriseServices.InsightsResolver = resolvers.New(rawInsightsDB, db)
 	enterpriseServices.CodeInsightsDataExportHandler = httpapi.NewExportHandler(db, rawInsightsDB).ExportFunc()
+	enterpriseServices.CompletionsResolver = resolvers2.NewCompletionsResolver()
```
Comment (novoselrok) [inline enterprise/cmd/frontend/internal/insights/init.go:43]: Why is the completions resolver initialized as a part of the insights resolvers?

File: cmd/frontend/graphqlbackend/completions.graphql [hunk 4]
```diff
@@ -0,0 +1,32 @@
+extend type Query {
+    """Returns a string of completion responses"""
+    completions(input: CompletionsInput!): String!
+}
+
+""" Input wrapper for completions """
```
Comment (novoselrok) [inline cmd/frontend/graphqlbackend/completions.graphql:6]: [Nitpick] comments in the rest of our GQL schemas are usually written as:

```
"""
Comment
"""
```

**Comment by mrnugget** [comment]
Hey! Since there's a lot of active thinking about APIs and versioning of them and where to move stuff -- how important is it to get this in? Or in other words: if this is in - do we need to keep it? migrate it to new stuff and keep it working?

File: cmd/frontend/graphqlbackend/completions.go (hunk 2 above)
Comment (coury-clark) [reply cmd/frontend/graphqlbackend/completions.go:3]: yep fixed, sometimes the autoimport goes wild in Goland

File: enterprise/cmd/frontend/internal/insights/init.go (hunk 3 above)
Comment (coury-clark) [reply enterprise/cmd/frontend/internal/insights/init.go:43]: accidentally pasted into the wrong file, fixed!

File: cmd/frontend/graphqlbackend/completions.graphql (hunk 4 above)
Comment (coury-clark) [reply cmd/frontend/graphqlbackend/completions.graphql:6]: yep thanks, I was trying to get away with 100% auto-generated comments 😄

**Review by novoselrok — APPROVED** [review]

**Pull request merged**